### PR TITLE
Catch more errors

### LINKED ISSUE #133: Write Wrapper Around Request Methods To Catch Various Errors
# Description
I noticed that we sometimes have an issue connecting to the game client API, which throws an uncaught ConnectionError.

The best fix for this is probably to write small wrapper methods for GET and POST to catch any errors in a more sustained way than through loguru.

At the moment I'm only thinking of catching ConnectionError, TimeoutError and HTTPError, but am open to handling more things.

Will probably implement this later today, but have to work first and didn't want to lose the thought :)

### LINKED ISSUE #134: Session Has Expired Is Not Dismissed
Just wanted to log this since I don't have time to fix it right now, I encountered this this morning after I took a peek at the bot:
image

Full verbose log: [tft-bot-debug-2023-04-12_00-30-32_212793.log](https://github.com/Kyrluckechuck/TFT-Bot/files/11211994/tft-bot-debug-2023-04-12_00-30-32_212793.log)

Relevant code snippet from the log that is repeated for thousands of lines:
```
2023-04-12 04:59:24.773 | DEBUG    | tft_bot.league_api.league_api_integration:in_game:283 - Checking if we are in a game
2023-04-12 04:59:24.778 | DEBUG    | tft_bot.league_api.league_api_integration:should_reconnect:305 - Checking if we should reconnect
2023-04-12 04:59:24.781 | DEBUG    | tft_bot.league_api.league_api_integration:is_dead:424 - Checking if we have more than 0 HP
2023-04-12 04:59:24.787 | INFO     | __main__:check_if_game_complete:434 - The game considers us dead, checking the screen again for exit buttons
2023-04-12 04:59:24.788 | INFO     | __main__:check_if_game_complete:419 - Waiting an additional ~25s for any exit buttons
2023-04-12 04:59:27.792 | INFO     | __main__:check_screen_for_exit_button:403 - End of game detected (exit now)
2023-04-12 04:59:31.432 | DEBUG    | tft_bot.helpers.click_helpers:click_to_middle:164 - M|Could not find 'captures/buttons/exit_now_highlighted.png', skipping
2023-04-12 04:59:33.340 | DEBUG    | __main__:check_screen_for_exit_button:405 - Exit now clicking success: True
2023-04-12 04:59:38.341 | INFO     | __main__:print_timer:694 - -----------------------------------------
2023-04-12 04:59:38.341 | INFO     | __main__:print_timer:695 - Game End
2023-04-12 04:59:38.342 | INFO     | __main__:print_timer:696 - Time since start: 4h 28m 23s
2023-04-12 04:59:38.344 | INFO     | __main__:print_timer:697 - Games played: 11
2023-04-12 04:59:39.098 | INFO     | __main__:print_timer:698 - Win rate (at most last 20 games): 72.73%
2023-04-12 04:59:39.098 | INFO     | __main__:print_timer:699 - -----------------------------------------
2023-04-12 04:59:39.100 | INFO     | __main__:match_complete:643 - Match complete! Cleaning up and restarting
2023-04-12 04:59:39.101 | DEBUG    | tft_bot.league_api.league_api_integration:in_game:283 - Checking if we are in a game
2023-04-12 04:59:40.862 | DEBUG    | tft_bot.league_api.league_api_integration:in_game:283 - Checking if we are in a game
2023-04-12 04:59:40.866 | DEBUG    | tft_bot.league_api.league_api_integration:in_queue:224 - Checking if we are already in a queue
2023-04-12 04:59:40.870 | INFO     | tft_bot.league_api.league_api_integration:create_lobby:186 - Creating a TFT lobby
2023-04-12 04:59:45.874 | DEBUG    | tft_bot.league_api.league_api_integration:in_game:283 - Checking if we are in a game
2023-04-12 04:59:45.877 | DEBUG    | tft_bot.league_api.league_api_integration:in_queue:224 - Checking if we are already in a
[..]
2023-04-12 09:28:07.515 | DEBUG    | tft_bot.league_api.league_api_integration:in_game:283 - Checking if we are in a game
2023-04-12 09:28:07.518 | DEBUG    | tft_bot.league_api.league_api_integration:in_queue:224 - Checking if we are already in a queue
2023-04-12 09:28:07.522 | INFO     | tft_bot.league_api.league_api_integration:create_lobby:186 - Creating a TFT lobby
2023-04-12 09:28:12.526 | DEBUG    | tft_bot.league_api.league_api_integration:in_game:283 - Checking if we are in a game
2023-04-12 09:28:12.529 | DEBUG    | tft_bot.league_api.league_api_integration:in_queue:224 - Checking if we are already in a queue
2023-04-12 09:28:12.532 | INFO     | tft_bot.league_api.league_api_integration:create_lobby:186 - Creating a TFT lobby
2023-04-12 09:28:17.536 | DEBUG    | tft_bot.league_api.league_api_integration:in_game:283 - Checking if we are in a game
2023-04-12 09:28:17.539 | DEBUG    | tft_bot.league_api.league_api_integration:in_queue:224 - Checking if we are already in a queue
2023-04-12 09:28:17.544 | INFO     | tft_bot.league_api.league_api_integration:create_lobby:186 - Creating a TFT lobby
2023-04-12 09:28:22.550 | DEBUG    | tft_bot.league_api.league_api_integration:in_game:283 - Checking if we are in a game
2023-04-12 09:28:22.553 | DEBUG    | tft_bot.league_api.league_api_integration:in_queue:224 - Checking if we are already in a queue
2023-04-12 09:28:22.558 | INFO     | tft_bot.league_api.league_api_integration:create_lobby:186 - Creating a TFT lobby
[me interfering to get the logs:]
2023-04-12 09:28:25.701 | DEBUG    | __main__:toggle_pause:151 - alt+p pressed, toggling pause from False to True
2023-04-12 09:28:25.701 | WARNING  | __main__:toggle_pause:154 - Bot now paused, remember to unpause to continue botting!
```

### PR DESCRIPTION
**Pull request by Gadsee**
# Description
This PR adds changes that catch some new errors that may occur since we switched to HTTP requests.
We now also catch KeyboardInterrupt to correctly display that we know we should exit instead of printing a random stack trace.
In addition, a fix has been added to check if our session has expired.

Fixes #133 
Fixes #134

### CODE REVIEW

File: tft.py [hunk 1]
```diff
@@ -893,4 +900,8 @@ def main():
 
 
 if __name__ == "__main__":
-    sys.exit(main())
+    try:
+        sys.exit(main())
+    except KeyboardInterrupt:
+        logger.info("Received wish to exit by CTRL+C, exiting immediately.")
```
Comment (Kyrluckechuck) [inline tft.py:906]: One nice QoL thing might be to just do a final `print_timer()` -- what do you think? 😛

File: tft.py [hunk 2]
```diff
@@ -219,6 +219,12 @@ def queue() -> None:  # pylint: disable=too-many-branches
         if not PLAY_NEXT_GAME:
             evaluate_next_game_logic()
 
+        if LCU_INTEGRATION.session_expired():
```
Comment (Kyrluckechuck) [inline tft.py:222]: Nice, much cleaner -- I didn't think we could pull that from the LCU.

**Review by Kyrluckechuck — COMMENTED** [review]
Looks great! Just one comment.

File: tft.py (hunk 1 above)
Comment (Gadsee) [reply tft.py:906]: Hm, you'd probably not see that unless you're running from source. In addition, the "intended" way to exit is to enable the "stop queueing" logic, and then to Ctrl+C or close the window, which would print it twice :/

File: tft.py (hunk 2 above)
Comment (Gadsee) [reply tft.py:222]: I'm using this great tool called LCUExplorer which lets you play around with the LCU through swagger, it's definitely worth a look if you ever have some time :) There's some really neat endpoints

File: tft.py (hunk 1 above)
Comment (Kyrluckechuck) [reply tft.py:906]: Fair enough -- I sometimes interrupt it mid-game and then surrender manually, so was thinking of that use case, but it's fairly minor 😛

**Review by Kyrluckechuck — APPROVED** [review]

File: tft.py (hunk 1 above)
Comment (Gadsee) [reply tft.py:906]: Hm, I think we can improve this catch to see from which logic step it was called, so maybe we can add it as a small improvement in another PR :)

**Pull request merged**